scrapinghub/client.py

```diff
@@ -247,16 +247,16 @@ def start(self, **params):
         return self._project.jobq.start(self, **params)
 
     def update(self, **params):
-        return self._project.jobq.update(self, **params)
+        return next(self._project.jobq.update(self, **params))['prevstate']
 
     def cancel(self):
         self._project.jobq.request_cancel(self)
 
     def finish(self, **params):
-        return self._project.jobq.finish(self, **params)
+        return next(self._project.jobq.finish(self, **params))['prevstate']
```

Comment:
Maybe I'm missing something, but if a) response is an empty list and b) `finish()` returns iterator wrapping it, then `next()` operation will raise StopIteration anyway. And if response is not empty, it will return a corresponding `prevstate`.
```
>>> a = iter([1])
>>> str(next(a))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```